pkg/vault/contracts/BatchRouter.sol

```diff
@@ -904,27 +910,27 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
             })
         );
 
-        if (EVMCallModeHelpers.isStaticCall()) {
-            return amountsOut;
-        }
+        bool isStaticCall = EVMCallModeHelpers.isStaticCall();
 
         for (uint256 i = 0; i < erc4626PoolTokens.length; ++i) {
             IERC4626 wrappedToken = IERC4626(address(erc4626PoolTokens[i]));
             IERC20 underlyingToken = IERC20(wrappedToken.asset());
 
-            // erc4626BufferWrapOrUnwrap will fail if the wrapper wasn't ERC4626
+            // erc4626BufferWrapOrUnwrap will fail if the wrapper is not ERC4626.
             (, , underlyingAmountsOut[i]) = _vault.erc4626BufferWrapOrUnwrap(
                 BufferWrapOrUnwrapParams({
                     kind: SwapKind.EXACT_IN,
                     direction: WrappingDirection.UNWRAP,
                     wrappedToken: wrappedToken,
-                    amountGivenRaw: amountsOut[i],
+                    amountGivenRaw: wrappedAmountsOut[i],
                     limitRaw: params.minAmountsOut[i],
                     userData: params.userData
                 })
             );
 
-            _sendTokenOut(params.sender, underlyingToken, underlyingAmountsOut[i], params.wethIsEth);
+            if (isStaticCall == false) {
```

Comment:
```suggestion
            if (isStaticCall == false) {
                IERC20 underlyingToken = IERC20(wrappedToken.asset());
```
`isStaticCall` is fine where it is (outside the loop, so it only gets called once), but moving this down could save a little gas on queries.